src/query/storages/fuse/fuse/src/io/segments.rs

```diff
@@ -74,22 +71,36 @@ impl SegmentsIO {
             }
         });
 
-        // 1.2 build the runtime.
-        let semaphore = Semaphore::new(max_io_requests);
-        let segments_runtime = Arc::new(Runtime::with_worker_threads(
-            max_runtime_threads,
-            Some("fuse-req-segments-worker".to_owned()),
-        )?);
+        try_join_futures(self.ctx.clone(), tasks).await
+    }
+}
 
-        // 1.3 spawn all the tasks to the runtime.
-        let join_handlers = segments_runtime.try_spawn_batch(semaphore, tasks).await?;
+pub async fn try_join_futures<Fut>(
+    ctx: Arc<dyn TableContext>,
+    futures: impl IntoIterator<Item = Fut>,
+) -> Result<Vec<Fut::Output>>
+where
+    Fut: Future + Send + 'static,
+    Fut::Output: Send + 'static,
+{
+    let max_runtime_threads = ctx.get_settings().get_max_threads()? as usize;
+    let max_io_requests = ctx.get_settings().get_max_storage_io_requests()? as usize;
 
-        // 1.4 get all the result.
-        let joint = future::try_join_all(join_handlers)
-            .instrument(tracing::debug_span!("read_segments_join_all"))
-            .await
-            .map_err(|e| ErrorCode::StorageOther(format!("read segments failure, {}", e)))?;
+    // 1. build the runtime.
+    let semaphore = Semaphore::new(max_io_requests);
+    let segments_runtime = Arc::new(Runtime::with_worker_threads(
+        max_runtime_threads,
+        Some("deletion-write-segments-worker".to_owned()),
```

Comment:
maybe better let caller site decides the thread name？

Reply:
> maybe better let caller site decides the thread name？

Added